pkg/components/ldap_auth.go

```diff
@@ -0,0 +1,69 @@
+package components
+
+import (
+	"context"
+	"net/http"
+)
+
+type authValidationTransport struct {
+	Wrapped              http.RoundTripper
+	AllowedGroups        []string
+	LdapGroupsHeaderName string
+}
+
+func Contains(s []string, target string) bool {
+	for _, c := range s {
+		if target == c {
+			return true
+		}
+	}
+	return false
+}
+
+func (r *authValidationTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	incomingLdapGroups := req.Header[r.LdapGroupsHeaderName]
+	allowedList := r.AllowedGroups
+	for _, g := range incomingLdapGroups {
+		if Contains(allowedList, g) {
+			continue
+		} else {
+			return newError(http.StatusUnauthorized, "missing required LDAP group"), nil
+		}
+	}
+	return r.Wrapped.RoundTrip(req)
+}
+
+// AuthConfig is used to configure authorization based on ldap group membership sent in a header
+type AuthConfig struct {
+	AllowedGroups        []string `description:"List of ldap groups allowed to access your service"`
+	LdapGroupsHeaderName string   `description:"Name of the header that contains the ldap group membership of an incoming request"`
+}
+
+// Name of the config root
+func (*AuthConfig) Name() string {
+	return "authconfig"
```

Comment:
task: the `config` part of this is redundant as everything is part of the config. This means configuration would look like:

```yaml
x-transportd:
  enabled:
    - authconfig
  authconfig:
    key: value
```

If we're going to call this `auth` then it should just be `auth`.